File: 21558 - Nithin/week 8/209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int n = nums.size();
        int min_len = INT_MAX;
        int sum = 0;
        int i,j = 0;

        for(i = 0;i <n;i++){
              sum += nums[i];
             
              while(sum >= target)
              {
                  min_len = min(min_len,(i+1)-j);
                  sum -= nums[j];
                  ++j;
              }
        }
        
        if(min_len == INT_MAX)
            return 0;
        
        return min_len;
    }
};

File: 21551-naveen-siva-kumar/week8/209_Minimum_Size Subarray Sum.py
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        
        int ans = INT_MAX;
        int windowS=0,windowE=0;
        int windowSize=0;
        int windowSum =0;
        for(int i=0;i<nums.size();i++){
            
            windowSum += nums[i];
            windowE++;
            
            while(windowSum>=target){
                ans = min(ans,windowE-windowS);
                windowSum -= nums[windowS];
                windowS++;
            }
        }
        return ans==INT_MAX?0:ans;
    }
};
